### ISSUE TITLE
CI (continuous integration) broken

### ISSUE THREAD

**Issue by bjendres** [Project60/org.project60.sepa#604]
It seems that something's wrong with the CI, we get:
```
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - cweagans/composer-patches is locked to version 1.6.5 and an update of this package was not requested.
    - cweagans/composer-patches 1.6.5 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.0.0] but it does not match the constraint.

You are using Composer 2, which some of your plugins seem to be incompatible with. Make sure you update your plugins or report a plugin-issue to ask them to support Composer 2.
```

Here's the full log: [ci.log](https://github.com/Project60/org.project60.sepa/files/5608940/ci.log)

Tests still pass on my local environment.

**Comment by bjendres** [comment]
@pfigel have you experienced this as well? I copied the CI from you :)

@michaelmcandrew I think we're using your docker container, does that ring a bell?

**Comment by michaelmcandrew** [comment]
@bjendres ah, that's interesting. as I understand it, the issue is that you are using composer 2 rather than composer 1. What image are you using? if it is something based on https://github.com/michaelmcandrew/civicrm-buildkit-docker/blob/master/civicrm/Dockerfile#L104 then as you will see from that line, all I am doing is running `civi-download-tools` so my assumption is that the problem is 'upstream' with buildkit.

If you are using https://github.com/michaelmcandrew/civicrm-docker/blob/master/templates/Dockerfile#L80 then I guess it is our problem and we should peg to composer 1.x for now since that is more civicrm friendly (which we will likely do in any case).

@elcapo fyi

**Comment by elcapo** [comment]
> ps @elcapo came across some similar Composer 2.x incompatibilities on another project just the other day.

The error message was about to be the same. @bjendres, can you try downgrading Composer and trying again:

```
composer self-update 1.4.1
composer update
```

You might need sudo privileges for the first command, depending on the context.

**Comment by bjendres** [comment]
Thanks @elcapo, but that doesn't quite seem to do the trick. I added ``composer self-update 1.4.1``  [HERE](https://github.com/Project60/org.project60.sepa/blob/issue/604/.circleci/config.yml#L20), then the you later run into

```
[...]
+ COMPOSER=composer
+ composer install
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file

                                                                                           
  [ErrorException]                                                                         
  "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?  
                                                                                           

install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--] [<packages>]...
```
(see [HERE](https://app.circleci.com/pipelines/github/Project60/org.project60.sepa/143/workflows/3443c666-47a0-4472-8c1a-6bf750c42244/jobs/405)).

When I add the second line
```
composer update
```
it breaks even earlier with 
```
Upgrading to version 1.4.1 (stable channel).
   
Use composer self-update --rollback to return to version 2.0.7
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Composer could not find a composer.json file in /root/project
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
```
Same thing happens if I add the parameters recommended by the URL (``composer --no-plugins --no-scripts update``).

@elcapo any ideas?

**Comment by pfigel** [comment]
@bjendres not sure, but I think this was due to compatibility issues between the PHP version in the CI image and composer 1.4.1. I've created a PR that uses the latest v1 and that seems to work.

**Comment by bjendres** [comment]
@pfigel fixed it!

**Comment by michaelmcandrew** [comment]
great - any thoughts on whether we should peg to composer 1 upstream? / I am not sure how long it will be before civicrm and friends become composer 2 friendly or if there are any other consderations.

**Comment by pfigel** [comment]
@michaelmcandrew good question - my understanding is that core on master and stable do okay with either composer 1 or 2. The CI failures were related to older Civi versions which we test against in addition to master/stable. I guess it would depend on how relevant that use-case (building older versions of Civi) is for the `civicrm-buildkit` image, beyond our CI thing.

**Comment by michaelmcandrew** [comment]
thanks @pfigel. I am coming to the conclusion that this is an upstream thing and a downstream thing but not the responsibility of civicrm-buildkit-docker. This image calls civi-download-tools for its dependencies and I think it makes sense for us to take our lead from there and not make any policy decisions that go against what they do. You tweaking downstream as you have done seems reasonable as well.

**Comment by jensschuppe** [comment]
@bjendres There's a branch `issue/604` lingering around (fiddling with CirceCI config) while this issue seems the have been resolved with #605 already. Can we delete the branch?

**Comment by bjendres** [comment]
> Can we delete the branch?

Yes
